pkg/cmd/kind/load/docker-image/docker-image.go

```diff
@@ -129,6 +136,16 @@ func runE(logger log.Logger, flags *flagpole, args []string) error {
 		}
 	}
 
+	if len(flags.ExcludeNodes) > 0 {
+		for _, name := range flags.ExcludeNodes {
+			_, ok := nodesByName[name]
+			if !ok {
+				return fmt.Errorf("unknown node: %q", name)
+			}
+			candidateNodes = nodeutils.RemoveNode(name, candidateNodes)
```

Comment:
the end result `candidateNodes` is processed with too many loops.
instead try using maps or a smarter way to process the excluded nodes.

for example, only add a node to candiateNodes if it's not in the excluded nodes.

Reply:
Thanks for your reply! Will update later.